### DISCUSSION TITLE
Can't render PDF despite PDFJS being enabled

### DISCUSSION THREAD

**Discussion by cashew22** [Igalia/cog#633] [Q&A] [answered]
Hi,

I'm trying to open a page like: https://www.orimi.com/pdf-test.pdf which is a simple PDF file. However, Cog seems to not be able to load pdfjs, it displayed the viewer.html code in the display instead of the actual PDF. I compared with Epiphany and we can see with the remote inspector that the two browser seems to behave differently.

My webkit is compiled with:
--  ENABLE_PDFJS                                    ON

there a screen shot of the webinspector downloaded content:
![image](https://github.com/Igalia/cog/assets/4152795/8188e727-ae02-46c3-b829-bcafea3fee4c)
Versus what I got on Epiphany
![image](https://github.com/Igalia/cog/assets/4152795/9df9dd46-0b0d-4ce1-8d19-a42520c3770e)

It seems that way more content is dowloaded.

I also have some message in the console, but it looks like it's in reaction of the fact that the pdfjs files are missing
![image](https://github.com/Igalia/cog/assets/4152795/77b74508-d837-4971-9894-59dc568a4ba5)

I'm open to suggestion if someone has an idea

Thanks

**Comment by cashew22** [comment, marked as answer]
Turn's out that you need to have the MIME database somewhere on the client side which is not a given when building from buildroot.

**Reply by aperezdc** [reply]
This is right, the `shared-mime-info` package needs to be installed to detect content types in a number of cases (e.g. as [stated here in the FAQ for local files](https://wpewebkit.org/about/faq.html#why-does-the-browser%2Flauncher-(e.g.-cog)-not-load-local-files%3F), and this is one of these cases.
